janus_server/src/aggregator.rs

```diff
@@ -3217,6 +3507,412 @@ mod tests {
+                start: Time(0),
+                duration: message::Duration(1000),
+            },
+            report_count: 5,
+            checksum: [0; 32],
+        };
+
+        let (parts, body) = warp::test::request()
+            .method("POST")
+            .path("/aggregate_share")
+            .body(AuthenticatedEncoder::new(request).encode(hmac_key))
+            .filter(&filter)
+            .await
+            .unwrap()
+            .into_response()
+            .into_parts();
+
+        assert_eq!(parts.status, StatusCode::BAD_REQUEST);
+        let problem_details: serde_json::Value =
+            serde_json::from_slice(&hyper::body::to_bytes(body).await.unwrap()).unwrap();
+        assert_eq!(
+            problem_details,
+            serde_json::json!({
+                "status": StatusCode::BAD_REQUEST.as_u16(),
+                "type": "urn:ietf:params:ppm:error:insufficientBatchSize",
+                "title": "There are not enough reports in the batch interval.",
+                "detail": "There are not enough reports in the batch interval.",
+                "instance": "..",
+                "taskid": base64::encode(task_id.as_bytes()),
+            })
+        );
+
+        // Interval is big enough, but checksum doesn't match
+        let request = AggregateShareReq {
+            task_id,
+            batch_interval: Interval {
+                start: Time(0),
+                duration: message::Duration(2000),
+            },
+            report_count: 10,
+            checksum: [3; 32],
+        };
+
+        let (parts, body) = warp::test::request()
+            .method("POST")
+            .path("/aggregate_share")
+            .body(AuthenticatedEncoder::new(request).encode(hmac_key))
+            .filter(&filter)
+            .await
+            .unwrap()
+            .into_response()
+            .into_parts();
+
+        assert_eq!(parts.status, StatusCode::BAD_REQUEST);
+        let problem_details: serde_json::Value =
+            serde_json::from_slice(&hyper::body::to_bytes(body).await.unwrap()).unwrap();
+        assert_eq!(
+            problem_details,
+            serde_json::json!({
+                "status": StatusCode::BAD_REQUEST.as_u16(),
+                "type": "urn:ietf:params:ppm:error:batchMisaligned",
+                "title": "The checksums or report counts in the two aggregator's aggregate shares do not match.",
+                "detail": "The checksums or report counts in the two aggregator's aggregate shares do not match.",
+                "instance": "..",
+                "taskid": base64::encode(task_id.as_bytes()),
+            })
+        );
+
+        // Interval is big enough, but report count doesn't match
+        let request = AggregateShareReq {
+            task_id,
+            batch_interval: Interval {
+                start: Time(0),
+                duration: message::Duration(2000),
+            },
+            report_count: 20,
+            checksum: [2; 32],
```

Comment:
This should be `3 ^ 2` as well, we may be taking the wrong branch to the right error.